include/ck/tensor_operation/gpu/element/element_wise_operation.hpp

```diff
@@ -114,29 +114,63 @@ struct AddHardswishAdd
 // E = FastGelu(C + D0 + D1)
 struct AddAddFastGelu
 {
+    // Fast GeLU
+    // https://paperswithcode.com/method/gelu
+    // y = 0.5*x*(1+tanh(sqrt(2/pi)*(x+0.044715*x^3)))
+    __host__ __device__ static constexpr float GetFastGeLU(float x)
+    {
+        const float u   = 2.f * x * (0.035677f * x * x + 0.797885f);
+        const float emu = exp(-u);
+        const float cdf = 0.5f + 0.5f * (2.f / (1.f + emu) - 1.f);
+        return x * cdf;
+    }
+
     template <typename E, typename C, typename D0, typename D1>
-    __host__ __device__ void operator()(E&, const C&, const D0&, const D1&) const;
+    __host__ __device__ constexpr void operator()(E&, const C&, const D0&, const D1&) const
+    {
+        static_assert(
+            always_false<E, C, D0, D1>,
+            "this generic version operator() is forbidden, please provide a valid specialization");
```

Comment:
I'm just curious is it really needed? The rest of the element_wise and binary operations didn't need that.

Reply:
The message is more confusing if we leave it as a declaration:  

```
warning: inline function 'ck::tensor_operation::element_wise::AddAddFastGelu::operator()<unsigned short, unsigned short, unsigned short, unsigned short>' is not defined [-W
undefined-inline]
```

Instead show warning and produce link error later, I think it would be better to prevent users to try instantiate this generic version and give reason, rather than allowing them to instantiate it somewhere.

```c++
error: static_assert failed due to requirement 'always_false<unsigned short, unsigned short, unsigned short, unsigned short>' "this generic version operator() is forbidden,
please provide a valid specialization"
```

Reply:
I will merge multiple specializations into single generic method and remove above-mentioned `static_assert()`.